### ISSUE TITLE
uint8 model runtime input(s) num is 2.

### ISSUE THREAD

**Issue by chenpengf0223** [tensorflow/tensorflow#41151]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- TensorFlow installed from (source or binary): binary
- TensorFlow version (or github SHA if from source): 2-1

**Command used to run the converter or code if you’re using the Python API**

```
import tensorflow as tf
from tensorflow.keras.models import load_model

keras_file = "./keras-0127127.h5"

model = load_model(keras_file)
converter = tf.lite.TFLiteConverter.from_keras_model(model)
converter.inference_input_type = tf.uint8
converter.inference_type = tf.uint8    #tf.lite.constants.QUANTIZED_UINT8
converter.inference_output_type = tf.uint8
converter.optimizations = [tf.lite.Optimize.OPTIMIZE_FOR_SIZE]
tflite_uint8_model = converter.convert()
open("uint8.tflite", "wb").write(tflite_uint8_model)
```

**The output from the converter invocation**
I use netron software to visualiz the "uint8.tflite" model, it shows that:
![image](https://user-images.githubusercontent.com/20535427/86765182-68cd0b00-c07b-11ea-83c2-026fbc5b5c50.png)
We can see some  DEPTHWISE_CONV_2D operations have 2 inputs.

**Also, please include a link to the keras model I used above**

```
https://github.com/chenpengf0223/semantic_segmentation_distillation/blob/master/keras-0127127.h5
```

**Failure details**
If the conversion is successful, but the generated model is wrong,
state what is wrong:
- Producing model that c++ tflite gpu delegate library can not process.

When I use c++ tflite gpu delegate library to run the model, I get such log:

INFO: Initialized TensorFlow Lite runtime.
INFO: Created TensorFlow Lite delegate for GPU.
ERROR: Following operations are not supported by GPU delegate:
DEPTHWISE_CONV_2D: Expected 1 runtime input tensor(s), but node has 2 runtime input(s).
DEQUANTIZE: Expected 1 runtime input tensor(s), but node has 0 runtime input(s).
20 operations will run on the GPU, and the remaining 41 operations will run on the CPU.
INFO: Initialized OpenCL-based API.
INFO: Created 1 GPU delegate kernels.

**Comment by multiverse-tf** [comment]
@srjoglekar246, could you take a look at this issue? It's related to quant model support in gpu delegate. But I'm not sure whether the root cause lies in the conversion.

**Comment by srjoglekar246** [comment]
@chenpengf0223 From the output, it looks like the model is partially running on GPU.

For some of the nodes, looks like the quantization isn't happening correctly. Could you try the following params for conversion:

```
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
converter.inference_input_type = tf.int8  # or tf.uint8
converter.inference_output_type = tf.int8  # or tf.uint8
converter.optimizations = [tf.lite.Optimize.DEFAULT]
```

(note: no `converter.inference_type`)

If that also leads to a similar output, there might be some issues with the quantization tooling - DEQUANTIZE nodes with no input tensors should never happen, so there might be some errors. Feel free to attach the tflite model as well, if things don't work after the above changes.

**Comment by chenpengf0223** [comment]
@srjoglekar246 Thanks for your reply, I try your paras as following:
converter.target_ops = [tf.lite.OpsSet.TFLITE_BUILTINS,tf.lite.OpsSet.SELECT_TF_OPS]
converter.inference_input_type = tf.int8  # or tf.uint8
converter.inference_output_type = tf.int8  # or tf.uint8
converter.optimizations = [tf.lite.Optimize.DEFAULT]

I also get a similar model, I attach the tflite model :
https://github.com/chenpengf0223/semantic_segmentation_distillation/blob/master/model-uint8.tflite

**Comment by srjoglekar246** [comment]
Do you mean the float model on GPU is faster than int8 model on GPU? Or that float model on CPU is faster than int8 on GPU?

If float model on GPU is faster than int8 model on GPU, could you share the float model also to debug?

There should not be such a big difference between float and int8.

**Comment by chenpengf0223** [comment]
@srjoglekar246  I run the float model and the int8 model on GPU using tflite gpu delegate, I attach the float32 model here:
https://github.com/chenpengf0223/semantic_segmentation_distillation/blob/master/float32.tflite

**Comment by srjoglekar246** [comment]
Looks like it is working as expected. 

For quantized models, we need to perform some extra operations to 'simulate' quantized inference on the GPU - this is done to ensure accuracy (compared to floating point).
This leads to increased latency, depending on characteristics such as number of input/output tensors, number of ops, etc.

Reducing the number of output tensors (6 in this case) will probably help.

However, it all boils down to what you want to trade-off: 
For very small model size but higher latency, choose int8. For 4x larger model but great speed, use float32. For something in between (slight accuracy tradeoff, 2x larger model) use [float16](https://www.tensorflow.org/lite/performance/post_training_float16_quant)
